shell/platform/android/platform_view_android.cc

```diff
@@ -317,7 +317,8 @@ std::unique_ptr<Surface> PlatformViewAndroid::CreateRenderingSurface() {
   if (!android_surface_) {
     return nullptr;
   }
-  return android_surface_->CreateGPUSurface();
+  return android_surface_->CreateGPUSurface(
+      android_context_->GetMainSkiaContext().get());
```

Comment:
This is the 2/2 of the actual active parts of this PR. If more GPUSurfaces are made with other AndroidSurfaces, re-use the same GrDirectContext from the AndroidContext.